portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -56,6 +86,22 @@ private String convertToJsonUsingGson(List messages) {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+    //TODO: Move into its own function.
+    //Populate the messages array.
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Task");
+    PreparedQuery results = datastore.prepare(query);
+
+    //Iterate results.
+    for (Entity entity : results.asIterable()) {
+
+        //Only add to array if it doesn't exist already.
+        if (!messages.contains(entity.getProperty("comment").toString())) {
```

Comment:
Does this mean you can't have the same comment text twice?

Like, if two people comment "I agree" on a post, only one of those will ever be retrieved?

Reply:
Fixed in latest commit!